apps/studio/src/utils/sitemap.ts

```diff
@@ -14,35 +20,69 @@ type ResourceDto = Omit<
   thumbnail?: string
 }
 
+const getResourcesWithFullPermalink = (
+  resources: ResourceDto[],
+  parent: ResourceDto,
+): ResourceDto[] => {
+  const children = resources
+    .filter((resources) => {
+      return parent.type === ResourceType.RootPage
+        ? resources.parentId === null
+        : resources.parentId === parent.id
+    })
+    .map((child) => {
+      return {
+        ...child,
+        permalink: `${parent.type === ResourceType.RootPage ? "" : parent.permalink}/${child.permalink}`,
+      }
+    })
+
+  return [
+    ...children,
+    ...children.flatMap((child) =>
+      getResourcesWithFullPermalink(resources, child),
+    ),
+  ]
+}
+
 const getSitemapTreeFromArray = (
   resources: ResourceDto[],
   parentId: string | null,
-  path: string,
 ): IsomerSitemap[] | undefined => {
   if (resources.length === 0) {
     return undefined
   }
 
   // Get the immediate children of the resource with the given parent ID
-  const children = resources.filter((resource) => {
-    if (parentId === null) {
+  const children = resources
+    .filter((resource) => {
+      const hasPageChildren = resources.some((possibleChild) => {
+        return (
+          possibleChild.permalink.startsWith(resource.permalink) &&
+          PAGE_RESOURCE_TYPES.find((t) => t === possibleChild.type)
+        )
+      })
       return (
-        resource.parentId === null &&
-        resource.type !== ResourceType.RootPage &&
-        resource.type !== ResourceType.FolderMeta &&
-        resource.type !== ResourceType.CollectionMeta
+        PAGE_RESOURCE_TYPES.find((t) => t === resource.type) || hasPageChildren
       )
-    }
-    return (
-      resource.parentId === parentId &&
-      resource.permalink !== INDEX_PAGE_PERMALINK
-    )
-  })
+    })
+    .filter((resource) => {
+      if (parentId === null) {
+        return (
+          resource.parentId === null &&
```

Comment:
nit: outside of where I can comment on this diff but I think we can remove the exclusion of `FolderMeta` and `CollectionMeta` here since it would have been excluded earlier under `PAGE_RESOURCE_TYPES`.

Reply:
uypdated the checks here to `resource.parentId === null`